src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -256,6 +258,25 @@ private Map<Integer, Float> doANNSearch(
             );
         }
 
+        QuantizationParams quantizationParams = quantizationService.getQuantizationParams(fieldInfo);
+
+        byte[] quantizedVector = null;
```

Comment:
This will not be the case always , It depends on type of Quantizer. It might be ok for BinaryQuantization but it wil change of others.

Reply:
We can change this as we add more types of quantization.  Eventually, we might have to run checks on the params and cast further down in the class.

Reply:
Lets add a TODO on top of this that this needs to be changed.